MathLib/LinAlg/Eigen/EigenOption.cpp

```diff
@@ -29,6 +32,10 @@ EigenOption::SolverType EigenOption::getSolverType(const std::string &solver_nam
         return SolverType::BiCGSTAB;
     if (solver_name == "SparseLU")
         return SolverType::SparseLU;
+#ifdef USE_EIGEN_UNSUPPORTED
+    if (solver_name == "GMRES")
+        return SolverType::GMRES;
```

Comment:
no, there is no reason and I prefer LIS. However currently those who download executables from the OGS website can use only Eigen solvers and they should be able to choose GMRES for unsymmetric matrices. BiCGSTab cannot be used for all the problems as it can break down.